pos_order_reorder/static/src/js/Screens/TicketScreen/ControlButtons/ReorderButton.js

```diff
@@ -0,0 +1,74 @@
+odoo.define("pos_order_reorder.ReorderButton", function (require) {
+    "use strict";
+
+    const {useListener} = require("@web/core/utils/hooks");
+    const PosComponent = require("point_of_sale.PosComponent");
+    const Registries = require("point_of_sale.Registries");
+    const {Orderline} = require("point_of_sale.models");
+
+    class ReorderButton extends PosComponent {
+        setup() {
+            super.setup();
+            useListener("click", this._onClick);
+        }
+        get isEmptyOrder() {
+            if (!this.props.order) return true;
+            return this.props.order.is_empty();
+        }
+        async _reOrder() {
+            if (this.isEmptyOrder) {
+                return;
+            }
+            const order = this.props.order;
+            const pos = this.env.pos;
+            const partner = order.get_partner();
+            const newOrder = pos.add_new_order();
+            if (partner) {
+                newOrder.set_partner(partner);
+            }
+            if (order.fiscal_position) {
+                newOrder.fiscal_position = order.fiscal_position;
+            }
+            if (order.pricelist) {
+                newOrder.set_pricelist(order.pricelist);
+            }
+            const lines = order.get_orderlines();
+            for (var i = 0; i < lines.length; i++) {
+                const line = lines[i];
+                const new_line = Orderline.create(
+                    {},
+                    {
+                        pos: pos,
+                        order: newOrder,
+                        product: pos.db.get_product_by_id(line.get_product().id),
+                        description: line.name,
+                        price: line.price_unit,
+                        tax_ids: newOrder.fiscal_position ? undefined : line.tax_id,
+                        price_manually_set: true,
+                        customer_note: line.customer_note,
+                    }
+                );
+                if (line.pack_lot_lines) {
+                    new_line.setPackLotLines({
+                        modifiedPackLotLines: [],
+                        newPackLotLines: (line.lot_names || []).map((name) => ({
+                            lot_name: name,
+                        })),
+                    });
+                }
+                new_line.set_unit_price(line.get_unit_price());
+                new_line.set_quantity(line.get_quantity());
+                new_line.set_discount(line.get_discount());
+                newOrder.add_orderline(new_line);
+            }
+            this.trigger("click-order", newOrder);
+        }
+        async _onClick() {
```

Comment:
The pile of execution starting from here is synchronous, you can remove all the `async` and `await` here, `_reOrder` should also get its `async` keyword removed.